### PR TITLE
Deprecate EventedControllerInterface

### LINKED ISSUE #7120: [Task] Check for deprecated stuff in method signatures
e.g. https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/lib/Controller/EventedControllerInterface.php#L25
--> FilterControllerEvent and FilterResponseEvent

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #7120 

## Additional info  
Deprecate `EventedControllerInterface` in favor of `KernelControllerEventInterface` and `KernelResponseEventInterface`

### CODE REVIEW

**Comment by brusch** [comment]
@dvesh3 looks good, I've just renamed `proxy` to `alias` since it isn't actually implementing the proxy pattern. ok for you?

**Comment by dvesh3** [comment]
> @dvesh3 looks good, I've just renamed `proxy` to `alias` since it isn't actually implementing the proxy pattern. ok for you?

yes, you are right. thanks for the changes 👍

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ If you have custom editables or wysiwyg global config then please change namespa
 - All `document_tag_` css classes are deprecated and will be removed in Pimcore 7. Use new classes `document_editable_` for custom styling of document editables.
 - Method signature `AbstractObject::setChildren($children)` has been updated to `AbstractObject::setChildren($children, array $objectTypes = [self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER], $includingUnpublished = false)`.
 - Edit Template for Area Bricks `edit.html.(php|twig)` has been deprecated and will be removed in Pimcore 7. Use new feature [editable dialog box](https://github.com/pimcore/pimcore/pull/6923#issuecomment-671257112) instead.
+- `EventedControllerInterface` is marked as deprecated and will be removed in v7. Please use new interfaces for kernel events KernelControllerEventInterface::onKernelControllerEvent() or KernelResponseEventInterface::onKernelResponseEvent instead.
```
Comment (jdreesen) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md:12]:
```suggestion
- `EventedControllerInterface` is marked as deprecated and will be removed in v7. Please use new interfaces for kernel events `KernelControllerEventInterface::onKernelControllerEvent()` or `KernelResponseEventInterface::onKernelResponseEvent()` instead.
```

Comment (dvesh3) [reply doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: thanks!

**Review by brusch — APPROVED** [review]

File: bundles/CoreBundle/EventListener/EventedControllerListener.php [hunk 2]
```diff
@@ -58,12 +66,21 @@ public function onKernelController(FilterControllerEvent $event)
     public function onKernelResponse(FilterResponseEvent $event)
     {
         $request = $event->getRequest();
-        $controller = $request->attributes->get('_evented_controller');
+        $eventedController = $request->attributes->get('_evented_controller');
+        $eventController = $request->attributes->get('_event_controller');
 
-        if (!$controller || !($controller instanceof EventedControllerInterface)) {
+        if (!$eventedController && !($eventController)) {
```
Comment (jdreesen) [inline bundles/CoreBundle/EventListener/EventedControllerListener.php:72]:
```suggestion
        if (!$eventedController && !$eventController) {
```

**Pull request merged**